parser/est/est.go

```diff
@@ -115,13 +115,21 @@ const (
 )
 
 type PubSubSubscriber struct {
-	Name     string   // The unique name of the subscriber
-	CallSite ast.Node // The AST node representing the creation of the subscriber
-	Func     ast.Node // The function that is the subscriber (either a *ast.FuncLit or a *ast.FuncDecl)
-	FuncFile *File    // The file the subscriber function is declared in
-	DeclFile *File    // The file that the subscriber is defined in
+	Name     string       // The unique name of the subscriber
+	Topic    *PubSubTopic // The topic the subscriber is registered against
+	CallSite ast.Node     // The AST node representing the creation of the subscriber
+	Func     ast.Node     // The function that is the subscriber (either a *ast.FuncLit or a *ast.FuncDecl)
+	FuncFile *File        // The file the subscriber function is declared in
+	DeclFile *File        // The file that the subscriber is defined in
+	IdentAST *ast.Ident   // The AST node representing the value this topic is bound against
 }
 
+func (p *PubSubSubscriber) Type() ResourceType         { return PubSubTopicResource }
+func (p *PubSubSubscriber) File() *File                { return p.DeclFile }
+func (p *PubSubSubscriber) Ident() *ast.Ident          { return p.IdentAST }
+func (p *PubSubSubscriber) NodeType() NodeType         { return PubSubSubscriberNode }
+func (p *PubSubSubscriber) AllowOnlyParsedUsage() bool { return true }
```

Comment:
We now have to implement the Resource interface, which requires an ident to be tracked.